src/coreclr/interpreter/compiler.cpp

```diff
@@ -2206,26 +2235,189 @@ InterpCompiler::InterpEmbedGenericResult InterpCompiler::EmitGenericHandle(CORIN
     return result;
 }
 
+InterpCompiler::GenericHandleData InterpCompiler::GenericHandleToGenericHandleData(const CORINFO_GENERICHANDLE_RESULT& embedInfo)
+{
+    if (embedInfo.lookup.lookupKind.needsRuntimeLookup)
+    {
+        CORINFO_RUNTIME_LOOKUP_KIND runtimeLookupKind = embedInfo.lookup.lookupKind.runtimeLookupKind;
+        InterpGenericLookup *pRuntimeLookup = (InterpGenericLookup*)AllocMethodData(sizeof(InterpGenericLookup));
```

Comment:
Is it the case that GenericHandleToGenericHandleData could get called for the same embedInfo multiple times, and we'd bloat the dataitems table with lots of duplicate lookups? i.e. if a method has two generic parameters T and U, are we going to end up with a unique dataitem for every use of T and every use of U? Or just two data items, one for T and one for U, that get reused through some other mechanism I haven't noticed yet?

Reply:
Yeah, this is a problem. I've added a general purpose hashing mechanism for embedding data into the dataitems array, and swapped over to using that hash even for regular pointers. This should address any reasonable scalability issues here.